src/http/server/handlers/websocket_handler.cr

```diff
@@ -9,45 +9,57 @@ require "http/web_socket"
   require "openssl/sha1"
 {% end %}
 
+# A handler which adds websocket functionality to an `HTTP::Server`.
+#
+# When a request can be upgraded, the associated `HTTP::Websocket` and
+# `HTTP::Server::Context` will be yielded to the block. For example:
+#
+# ```
+# ws_handler = HTTP::WebSocketHandler.new do |ws, ctx|
+#   ws.on_ping { ws.pong ctx.request.path }
+# end
+# server = HTTP::Server.new [ws_handler]
+# ```
 class HTTP::WebSocketHandler
   include HTTP::Handler
 
   def initialize(&@proc : WebSocket, Server::Context ->)
   end
 
   def call(context)
-    if websocket_upgrade_request? context.request
-      response = context.response
-
-      version = context.request.headers["Sec-WebSocket-Version"]?
-      unless version == WebSocket::Protocol::VERSION
-        response.status = :upgrade_required
-        response.headers["Sec-WebSocket-Version"] = WebSocket::Protocol::VERSION
-        return
-      end
-
-      key = context.request.headers["Sec-WebSocket-Key"]?
-
-      unless key
-        response.respond_with_status(:bad_request)
-        return
-      end
-
-      accept_code = WebSocket::Protocol.key_challenge(key)
-
-      response.status = :switching_protocols
-      response.headers["Upgrade"] = "websocket"
-      response.headers["Connection"] = "Upgrade"
-      response.headers["Sec-WebSocket-Accept"] = accept_code
-      response.upgrade do |io|
-        ws_session = WebSocket.new(io, sync_close: false)
-        @proc.call(ws_session, context)
-        ws_session.run
-      ensure
-        io.close
-      end
-    else
-      call_next(context)
+    unless websocket_upgrade_request? context.request
+      call_next context
+      return
```

Comment:
```suggestion
      return call_next(context)
```

Reply:
I was following the pattern set out below by putting the return on a separate line

Reply:
I think the return in the same line is fine. I was just taking about not inlining the unless.

Reply:
This also keeps the previous behaviour.

Reply:
fixed, and changed the other to be consistent with this in that case

Reply:
@dscottboggs actually, the other case should be left as-is.